components/pd_client/src/client.rs

```diff
@@ -146,18 +151,25 @@ impl RpcClient {
 
     /// Gets the leader of PD.
     pub fn get_leader(&self) -> Member {
-        self.leader_client.get_leader()
+        self.pd_client.get_leader()
     }
 
     /// Re-establishes connection with PD leader in synchronized fashion.
     pub fn reconnect(&self) -> Result<()> {
-        block_on(self.leader_client.reconnect(true))
+        block_on(self.pd_client.reconnect(true))
     }
 
     /// Creates a new call option with default request timeout.
     #[inline]
-    fn call_option() -> CallOption {
-        CallOption::default().timeout(Duration::from_secs(REQUEST_TIMEOUT))
+    fn call_option(client: &Client) -> CallOption {
+        let mut builder = MetadataBuilder::with_capacity(1);
```

Comment:
Better extracting to a standalone function for setting metadata and should skip if `forwarded_host` is empty.